App/Services/Notifications.ts

```diff
@@ -9,23 +9,65 @@ export interface INotificationsPayload {
   typeString: string,
 }
 
-export function isPhoto(notification: Notification): boolean {
-  if (!notification.data_id) {
-    return false
-  }
-  switch (notification.type) {
-    case NotificationType.receivedInviteNotification:
+export function toTypedNotification(notificationData: NotificationData): Notification {
+  const { subject_id, subject, data_id, type, ...baseNotification } = notificationData
+  switch (type) {
+    case NotificationType.commentAddedNotification:
+      return {
+        ...baseNotification,
+        type,
+        threadId: notificationData.subject_id as any,
```

Comment:
Any time you have a trusted normal value (`string` in this case) that you need to convert to the nominal type, you use `as any`.